### PR TITLE
Fix package/dependency creation bugs

### PR DESCRIPTION
**Pull request by AyanSinhaMahapatra**
### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

File: src/packagedcode/pypi.py [hunk 1]
```diff
@@ -71,6 +71,14 @@ class BasePypiHandler(models.DatafileHandler):
     def compute_normalized_license(cls, package):
         return compute_normalized_license(package.declared_license)
 
+    @classmethod
```
Comment (pombredanne) [inline src/packagedcode/pypi.py]: What do you gain from overriding these two methods with the exact same code that's in the super class?

**Review by pombredanne — APPROVED** [review]
LGTM!
I just have a small comment wrt. https://github.com/nexB/scancode-toolkit/pull/2932/files#r861907921 where subclassing is not needed especially since you call super() which will travel up the class hierarchy and MRO alright

File: src/packagedcode/pypi.py (hunk 1 above)
Comment (AyanSinhaMahapatra) [reply src/packagedcode/pypi.py]: If in pypi the `PythonEditableInstallationPkgInfoFile.assign_package_to_resources` is overridden to call `cls.assign_package_to_parent_tree` then, the `assign_package_to_parent_tree` again calls the `assign_package_to_resources` function which has now been overridden to call the same. This causes the recursion.

**Comment by AyanSinhaMahapatra** [comment]
@pombredanne updated that.

**Comment by pombredanne** [comment]
LGTM. Thanks! merging!

**Pull request merged**